tests/MODULE.bazel

```diff
@@ -1,6 +1,156 @@
-###############################################################################
-# Bazel now uses Bzlmod by default to manage external dependencies.
-# Please consider migrating your external dependencies from WORKSPACE to MODULE.bazel.
-#
-# For more details, please check https://github.com/bazelbuild/bazel/issues/18958
-###############################################################################
+module(name = "rules_scala_annex_tests")
+
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
```

Comment:
I think we can mark some of these as `dev_dependency`, but I'm not sure it matters here because no one's going to depend on the test workspace.

Reply:
We could, but yeah, it wouldn't benefit us at all. And if someone did depend on `rules_scala_annex_tests`, it would be incorrect for the reason I describe here:
https://github.com/lucidsoftware/rules_scala/pull/74#discussion_r1873510258